### PR TITLE
fix(merge-tree): Tolerate local references without segments in SortedSegmentSet

### PR DESCRIPTION
**Pull request by Abe27342**
## Description

Removes a bad assumption in `SortedSegmentSet` that local references' segments are defined. This code does not actually require the segment, just that any local reference with an undefined segment reports a consistent ordering key.

### CODE REVIEW

**Comment by Abe27342** [comment]
needs a regression test added, just opening this now to get tests run on it

File: packages/dds/merge-tree/src/test/client.localReference.spec.ts [hunk 1]
```diff
@@ -629,81 +630,112 @@ describe("MergeTree.Client", () => {
...
-		segInfo.segment.localRefs.removeLocalRef(localRef);
-		assert(segInfo.segment.localRefs.empty);
-		(localRef as any).link(undefined, 0, undefined);
-		assert(segInfo.segment.localRefs.empty);
-
-		assert.equal(segInfo.segment.localRefs.empty, true);
-		assert.equal(segInfo.segment.localRefs.has(localRef), false);
-		assert.equal(localRef.getSegment(), undefined);
-		assert.equal(localRef.getOffset(), 0);
+	const tgCases = [
```
Comment (connorskees) [inline packages/dds/merge-tree/src/test/client.localReference.spec.ts:633]: Does this repro the crash, or just adding more coverage?

Comment (Abe27342) [reply packages/dds/merge-tree/src/test/client.localReference.spec.ts:633]: it reproduces the issue reported for the `TrackingGroup` case. The other cases passed before (but still good to check)

**Review by connorskees — APPROVED** [review]
this looks good to me, though maybe @anthony-murphy would want to take a look

**Pull request merged**